code/game/objects/items/mail.dm

```diff
@@ -332,3 +334,166 @@
+		// If the recipient's mind has gone, then anyone can open their mail
+		// whether a mind can actually be qdel'd is an exercise for the reader
+		if(recipient && recipient != user?.mind)
+			to_chat(user, span_notice("You can't open somebody else's mail! That's <em>illegal</em>!"))
+			return
+
+	to_chat(user, span_notice("You start to unwrap the package..."))
+	if(!do_after(user, 1.5 SECONDS, target = user))
+		return
+	user.temporarilyRemoveItemFromInventory(src, TRUE)
+	playsound(loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+	for(var/obj/item/stuff in contents) // Mail and envelope actually can have more than 1 item.
+		if(user.put_in_hands(stuff) && armed)
+			log_bomber(user, "opened armed mail made by [madeby] ([madeby.ckey]), activating", stuff)
+			stuff.attack_self(user)
+	qdel(src)
+
+/obj/item/mail/traitor/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(armed)
+		if(W.tool_behaviour == TOOL_MULTITOOL)
+			to_chat(user, span_notice("You start to disable something..."))
+			
+			if(do_after(user, 2 SECONDS, target = src))
+				to_chat(user, span_notice("You have disarmed some kind of device..."))
+				playsound(src, 'sound/machines/defib_ready.ogg', 50, TRUE)
+				armed = FALSE
+
+/obj/item/storage/mail_counterfeit_device
+	name = "mail counterfeit device"
+	desc = "Device that actually able to counterfeit NT's mail. This device also able to place a trap inside of mail for malicious actions. Trap will \"activate\" any item inside of mail. Also it might be used for contraband purposes. Integrated micro-computer will give you great configuration optionality for your needs."
+	w_class = WEIGHT_CLASS_NORMAL
+	icon = 'icons/obj/device_syndie.dmi'
+	icon_state = "mail_counterfeit_device" // placeholder for now
+
+/obj/item/storage/mail_counterfeit_device/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 1
+	atom_storage.allow_big_nesting = TRUE
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+
+/obj/item/storage/mail_counterfeit_device/attack_self(mob/user, modifiers)
+	var/mail_type = tgui_alert(user, "Is it gonna be an envelope or a normal mail?", "Mail Counterfeiting", list("Mail", "Envelope"))
+	if(isnull(mail_type))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+	mail_type = lowertext(mail_type)
+
+	var/mail_armed = tgui_alert(user, "Is it gonna be armed?", "Mail Counterfeiting", list("Yes", "No"))
+	if(isnull(mail_armed))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+
+	if(mail_armed == "Yes")
+		mail_armed = TRUE
+	else
+		mail_armed = FALSE
+
+	var/list/mail_recipients = list()
+	var/list/mail_recipients_input_list = list("0# Anyone")
+	var/iterator = 1
+	for(var/mob/living/carbon/human/human in GLOB.human_list)
+		// Skip everyone who is not part of the crew...
+		if(isnull(human.mind))
+			continue
+		if(!(human.mind.assigned_role.job_flags & JOB_CREW_MEMBER))
+			continue
+		mail_recipients += human.mind
+		mail_recipients_input_list += "[iterator]# [human.name]"
```

Comment:
Also wait, this is going to leak a lot of information if we're only doing it based off actively existing mobs (gibbed mobs won't show up), and using `name` will update based on what their disguise is, which we don't want! You should be looking through the crew manifest.

Reply:
i see...

Reply:
From there duplicates are going to be so rare as to be non-existent, and you can safely either exclude them from the list or handle them with (2) etc if they are duplicated.

Reply:
I made it bound to the manifest. But i will leave iterators, the reason is [here](https://github.com/tgstation/tgstation/pull/75390#discussion_r1192704961).

Reply:
Do not leave the iterators, they are hideous. Only include (2), (3) etc if necessary.

Reply:
IIRC we have a helper for sorting a list and including (1) (2) (3) if there are dupes. Heretic curses use it and it does something extremely similar to this